src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Search/SortValueJObjectGenerator.cs

```diff
@@ -0,0 +1,101 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.CosmosDb.Features.Search;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage.Search
+{
+    public class SortValueJObjectGenerator : ISearchValueVisitor
+    {
+        private string _prefix;
+
+        private JObject CurrentEntry { get; set; }
+
+        public JObject Generate(SortValue entry)
+        {
+            EnsureArg.IsNotNull(entry, nameof(entry));
+
+            CreateEntry();
+
+            if (entry.Low != null)
+            {
+                _prefix = SearchValueConstants.SortLowValueFieldName;
+                entry.Low.AcceptVisitor(this);
+            }
+            else
+            {
+                AddProperty(SearchValueConstants.SortLowValueFieldName, null);
+            }
+
+            if (entry.Hi != null)
+            {
+                _prefix = SearchValueConstants.SortHighValueFieldName;
+                entry.Hi.AcceptVisitor(this);
+            }
+            else
+            {
+                AddProperty(SearchValueConstants.SortHighValueFieldName, null);
+            }
+
+            return CurrentEntry;
+        }
+
+        public void Visit(CompositeSearchValue composite)
+        {
+        }
+
+        public void Visit(DateTimeSearchValue dateTime)
+        {
+            AddProperty(_prefix, dateTime.Start.ToString("o", CultureInfo.InvariantCulture));
```

Comment:
Should this be` dateTime.End` when _prefix is `SearchValueConstants.SortHighValueFieldName`?

Reply:
Thanks, good call. I reverted this when I changed to a single field